### DISCUSSION TITLE
Any duckdb <> sqlite bridging vtab implementations?

### DISCUSSION THREAD

**Discussion by hoehrmann** [duckdb/duckdb#6776] [Q&A]
I am very familiar with the virtual table mechanism in SQLite and understand that duckdb provides a similar mechanism. Are there any implementations of bridges between them? I am aware there is a duckdb extension to load SQLite databases, but what I have in mind would also provide access to SQLite virtual tables from duckdb (and vice versa), i.e., an application would have to properly host both databases so their extensions can be properly loaded and you actually have access to the full range of their combined features. As an example, I have an application that makes extensive use of SQLite features and custom virtual tables. But I might have queries where duckdb supports the needed features, so I might want to run duckdb queries on SQLite tables / virtual tables / views. From a cursory glance over [](https://duckdb.org/docs/api/c/table_functions) my impression is that it should be fairly simple to make table functions that simply proxy data between duckdb and SQLite. But it would seem odd that I should be the first person to be interested in such a functionality. Web search did not bring up much though.

**Comment by Alex-Monahan** [comment]
Hello! I believe the `ATTACH` command will work perfectly for what you need! Could you give it a try and let us know how it goes?
https://duckdb.org/docs/sql/statements/attach
